### PR TITLE
NTP-1158: Transition all pricing to ex.VAT

### PR DESCRIPTION
**Pull request by kieranohanlon**
## Context

Transition all pricing to ex.VAT

## Changes proposed in this pull request

Update the TP details page so shows exc and inc VAT and the compare to be defaulted to exc VAT on load

## Guidance to review

Test that the VAT is shown correctly

## Link to Jira ticket

https://dfedigital.atlassian.net/browse/NTP-1158

## Things to check

- [x] Title is in the format "NTP-XXX: Short description" where NTP-XXX is the Jira ticket reference
- [x] Code and tests follow the [coding standards](/docs/coding-standards.md)
- [x] `dotnet format` has been run in the repository root
- [ ] Test coverage of new code is at least 80%
- [x] All UI related acceptance criteria specified in the ticket have been added to the relevant [cypress test feature file(s)](/UI/cypress/e2e/)
- [ ] Unless explicitly mentioned in the ticket, all UI work should have been tested as working on desktop and mobile resolutions with JavaScript on and off
- [ ] Database migrations can be applied to the current codebase in main without causing exceptions
- [ ] Debug logging has been added after all logic decision points
- [x] All [automated testing](/README.md#testing) including accessibility and security has been run against your code
- [x] **PR deployment has been signed off by wider team**

### CODE REVIEW

File: UI/Pages/TuitionPartner.cshtml [hunk 1]
```diff
@@ -3,305 +3,392 @@
...
-    <span show-if="!@Model.Data.HasPricingVariation" data-testid="pricing-same-for-subjects">
-        All subjects are the same price but depend on tuition group size.
-    </span>
-    <span show-if="@Model.Data.HasPricingVariation" data-testid="pricing-differences-for-subjects">
-        Some prices are different depending on key stage or subject. Please contact @Model.Data.Name for more information.
-    </span>
-    <span show-if="@Model.Data.IsVatCharged" data-testid="price-includes-vat"><br/>Prices shown include VAT.</span>
-    <span show-if="@Model.Data.IsVatCharged==false" data-testid="price-vat-not-applicable"><br/>VAT does not apply to these prices because @Model.Data.Name is VAT exempt.</span>
-</p>
+      @if (!Model.Data.AllPrices.Any())
```
Comment (hasan3d) [inline UI/Pages/TuitionPartner.cshtml:138]: why do we need to check the `!Model.Data.AllPrices.Any()`?

File: UI/Pages/TuitionPartner.cshtml [hunk 2]
```diff
@@ -3,305 +3,392 @@
...
+        </span>
+        <span show-if="@Model.Data.HasPricingVariation" data-testid="pricing-differences-for-subjects">
+          Some prices are different depending on key stage or subject. Please contact @Model.Data.Name for more information.
+        </span>
+        @if (!Model.Data.IsVatCharged)
+        {
+          <div class="govuk-inset-text">
+            <p class="govuk-body">This tuition partner does not charge VAT.</p>
+            <p class="govuk-body">Tuition partners such as charities or Limited Liability Partnerships (LLP) do not charge VAT.</p>
+          </div>
```
Comment (hasan3d) [inline UI/Pages/TuitionPartner.cshtml:134]: We can add the logic for the one price table when a TP does not charge VAT here:

```
        <table class="govuk-table" data-testid="pricing-table">
          <caption class="govuk-visually-hidden">Tuition prices (excluding VAT)</caption>
          <thead class="govuk-table__head">
            <tr class="govuk-table__row">
              <th scope="col" class="govuk-table__header">Tuition group size</th>
              <th show-if="@Model.Data.Prices.ContainsInSchoolPrice()" scope="col" class="govuk-table__header">In school</th>
              <th show-if="@Model.Data.Prices.ContainsOnlinePrice()" scope="col" class="govuk-table__header">Online</th>
            </tr>
          </thead>
          <tbody class="govuk-table__body">
            @foreach (var item in Model.Data.Prices)
            {
              <tr class="govuk-table__row">
                <th scope="row" class="govuk-table__header" data-testid="pricing-group-size-column">
                  @(
                      ((Domain.Enums.GroupSize)item.Key).DisplayName()
                      )
                </th>
                <td show-if="@Model.Data.Prices.ContainsInSchoolPrice()" class="govuk-table__cell">
                  @item.Value.FormatFor(Domain.Enums.TuitionType.InSchool, false)
                </td>

                <td show-if="@Model.Data.Prices.ContainsOnlinePrice()" class="govuk-table__cell">
                  @item.Value.FormatFor(Domain.Enums.TuitionType.Online, false)
                </td>
              </tr>
            }
          </tbody>
        </table>
```

File: UI/Pages/TuitionPartner.cshtml [hunk 3]
```diff
@@ -3,305 +3,392 @@
...
+                  {
+                    <th scope="col" class="govuk-table__header">@subjectName</th>
+                  }
+                </tr>
+              </thead>
+              <tbody class="govuk-table__body">
+                @for (var groupSize = 1; groupSize < 7; groupSize++)
+                {
+                  <tr class="govuk-table__row">
+                    <th scope="row" class="govuk-table__header">@(((Domain.Enums.GroupSize)groupSize).DisplayName())</th>
```
Comment (hasan3d) [inline UI/Pages/TuitionPartner.cshtml:261]: The `Domain.Enums` Qualifier is redundant for the `Domain.Enums.GroupSize.` We already have a using statement on that file for this.

File: UI/Pages/TuitionPartner.cshtml (hunk 2 above)
Comment (kieranohanlon) [reply UI/Pages/TuitionPartner.cshtml:134]: I'm not sure that I understand this one?  The scenario where the TP does not charge VAT is already catered for, with the following code or was the comment meaning something else:

```
      @if (!Model.Data.AllPrices.Any())
      {
        <h3 class="govuk-heading-s" show-if="@Model.Data.IsVatCharged">Tuition prices (excluding VAT)</h3>
        <table class="govuk-table" data-testid="pricing-table">
...
```

File: UI/Pages/TuitionPartner.cshtml (hunk 3 above)
Comment (kieranohanlon) [reply UI/Pages/TuitionPartner.cshtml]: This was copied from existing code, but is a good spot, I've removed all the redundant `Domain.Enum.`

File: UI/Pages/TuitionPartner.cshtml (hunk 2 above)
Comment (hasan3d) [reply UI/Pages/TuitionPartner.cshtml:134]: Great, I understand what you're saying. By default, we only display the title if a TP charges VAT for the pricing table. However, if a TP does charge VAT, we display both the including and excluding prices, along with the title for the table.

**Review by hasan3d — APPROVED** [review]
lgtm

**Pull request merged**